src/Controller/Controller.php

```diff
@@ -851,17 +851,13 @@ public function paginate(
                 $settings
             );
         } catch (PageOutOfBoundsException $exception) {
-            // Nothing to do here, NotFoundException will be thrown below.
+            throw new NotFoundException(null, null, $exception);
```

Comment:
Should we just throw the PageOutOfBounds exception?

Reply:
Can't, because it doesn't extend HttpException and we don't want a 500 error.

Reply:
I guess I was asking if we want to change that.

Reply:
Want maybe, but that would add Http package as a dependency for the datasource package, which we don't want.

Reply:
oh, right, that old nonsense. yea I'm not looking to add many more exception code overrides to the renderer either.